config/formats.ts

```diff
@@ -2,7 +2,79 @@
 // The rules that formats use are stored in data/rulesets.js
 import {Utils} from './../lib/utils';
 
-export const Formats: (FormatsData | {section: string, column?: number})[] = [
+import { SideFormats } from './custom-formats';
+
+// interface for the builder.
+interface formatSection {
+	section: string
+	column?: number
+	formats: FormatsData[];
+}
+
+// function for merging the two lists
+function merge(main: (FormatsData | {section: string, column?: number})[],
+			   side: (FormatsData | {section: string, column?: number})[]):
+			   (FormatsData | {section: string, column?: number})[] {
+	// result that is return and makes the actual list for formats.
+	var result: (FormatsData | {section: string, column?: number})[] = [];
+
+	// used as a intermediary to build the final list.
+	var build: formatSection[] = [];
+
+	// used to track location to keep formats under their sections.
+	var loc: number = -1;
```

Comment:
```suggestion
	const loc: number = -1;
```

Reply:
This isn't const though. I change it constantly.

Reply:
use `let` then

Reply:
will do, thank you.